crates/runtime/src/dataconnector/graphql.rs

```diff
@@ -79,20 +78,98 @@ Please verify the syntax of your GraphQL query."#
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
-pub struct GraphQL {
-    secret: Option<Secret>,
-    params: Arc<HashMap<String, String>>,
+#[derive(Debug, PartialEq, Eq)]
+struct PaginationParameters {
+    resource_name: String,
+    count: usize,
+    page_info_path: String,
 }
 
-impl DataConnectorFactory for GraphQL {
-    fn create(
-        secret: Option<Secret>,
-        params: Arc<HashMap<String, String>>,
-    ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
-        Box::pin(async move {
-            let graphql = Self { secret, params };
-            Ok(Arc::new(graphql) as Arc<dyn DataConnector>)
-        })
+impl PaginationParameters {
+    fn parse(query: &str, pointer: &str) -> Option<Self> {
+        let pagination_pattern = r"(?xsm)(\w*)\s*\([^)]*first:\s*(\d+).*\).*\{.*pageInfo.*\{.*hasNextPage.*endCursor.*?\}.*?\}";
+        let regex = unsafe { Regex::new(pagination_pattern).unwrap_unchecked() };
```

Comment:
Why we are using `unsafe` + `unwrap_unchecked`? Because `pagination_pattern` is hard-coded string and we know it works?

Reply:
Let's use `let regex = Regex::new(pagination_pattern).expect("Invalid regex pagination pattern")` instead. It will still panic in case of an error but with more details.

On the topic of graceful/panic error handling I like [this answer](https://stackoverflow.com/a/39478185). TLDR:
- if error is recoverable, we should handle it gracefully
- if error is a bug, it is ok to panic

I think we need to put that in our guidelines cc @phillipleblanc

Reply:
Used panic! instead of unwrap_unchecked